src/sagemaker/jumpstart/session_utils.py

```diff
@@ -0,0 +1,159 @@
+
+    Raises:
+        ValueError: If model ID and version cannot be inferred from the endpoint.
+    """
+    if inference_component_name or sagemaker_session.is_ic_based_endpoint(endpoint_name):
+        if inference_component_name:
+            model_id, model_version = _get_model_id_version_from_ic_endpoint_with_ic_name(
+                inference_component_name, sagemaker_session
+            )
+
+        else:
+            (
+                model_id,
+                model_version,
+                inference_component_name,
+            ) = _get_model_id_version_from_ic_endpoint_without_ic_name(
+                endpoint_name, sagemaker_session
+            )
+
+    else:
+        model_id, model_version = _get_model_id_version_from_non_ic_endpoint(
+            endpoint_name, inference_component_name, sagemaker_session
+        )
+    return model_id, model_version, inference_component_name
+
+
+def _get_model_id_version_from_ic_endpoint_without_ic_name(
+    endpoint_name: str, sagemaker_session: Session
+) -> Tuple[str, str, str]:
+    """Given an endpoint name, derives the model ID, version, and inferred inference component name.
+
+    This function assumes the endpoint corresponds to an inference component-based endpoint.
+
+    Raises:
+        ValueError: If there is not a single inference component associated with the endpoint.
+    """
+    inference_components = sagemaker_session.list_inference_components(
+        endpoint_name_equals=endpoint_name
+    )["InferenceComponents"]
+
+    if len(inference_components) == 0:
+        raise ValueError(
+            f"No inference components found for the following endpoint: {endpoint_name}. "
+            "Use ``SageMaker.CreateInferenceComponent`` to add inference components to "
+            "your endpoint."
+        )
+    if len(inference_components) > 1:
+        raise ValueError(
+            "Must provide inference component name for endpoint with "
+            "multiple inference components."
+        )
+    inference_component_name = inference_components[0]["InferenceComponentName"]
+    return (
+        *_get_model_id_version_from_ic_endpoint_with_ic_name(
+            inference_component_name, sagemaker_session
+        ),
+        inference_component_name,
+    )
+
+
+def _get_model_id_version_from_ic_endpoint_with_ic_name(
+    inference_component_name: str, sagemaker_session: Session
+):
+    """Returns the model ID and version inferred from a sagemaker inference component.
+
+    Raises:
+        ValueError: If the inference component does not have tags from which the model ID
+            and version can be inferred.
+    """
+    region: str = sagemaker_session.boto_region_name
+    partition: str = aws_partition(region)
+    account_id: str = sagemaker_session.account_id()
+
+    inference_component_arn = (
+        f"arn:{partition}:sagemaker:{region}:{account_id}:"
+        f"inference-component/{inference_component_name}"
```

Comment:
Quick note, IC is the only entity that we return mix-cased ARNs to customers in DescribeIC/ListICs. Backend we store lower-cased ARNs as DDB primary key.

Existing Model, EndpointConfig, Endpoint entities haven't completed ARN case sensitivity migration yet, so customers will see lower-cased ARNs for these 3 in their Describe*/List*.

Reply:
thanks for the note. Do i need to lowercase the arn then?

Reply:
no Evan, that means you can keep the inference component name as is in the ARN. However for other hosting resources (Model, EndpointConfig, Endpoint) we must remember to lower case the endpoint name before constructing the ARN.